modules/data_release/ajax/FileUpload.php

```diff
@@ -12,36 +12,37 @@
  * @link     https://github.com/aces/Loris
  */
 
-$DB   = \Database::singleton();
-$user = \User::singleton();
+$factory = \NDB_Factory::singleton();
+$DB      = $factory->database();
+$user    = $factory->user();
+$config  = $factory->config();
 
 if ($_POST['action'] == 'upload'
     && $user->hasPermission("data_release_upload")
 ) {
     $fileName    = $_FILES["file"]["name"];
     $version     = $_POST['version'];
     $upload_date = date('Y-m-d');
-    $base_path   = __DIR__ . "/../user_uploads/";
 
-    $factory  = NDB_Factory::singleton();
     $settings = $factory->settings();
 
     $baseURL = $settings->getBaseURL();
+    $path    = $config->getSetting('dataReleasePath');
 
-    if (!file_exists(__DIR__ . "/../user_uploads/")) {
+    if (!file_exists($path)) {
         error_log(
-            "ERROR: File upload failed. Default user_uploads"
+            "ERROR: File upload failed. Default upload"
```

Comment:
```suggestion
            "ERROR: File upload failed. Upload"
```

I think this error should be more general because it doesn't really matter whether the directory is default or not if it isn't found.

Reply:
right